Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -562,13 +562,16 @@ namespace Babylon
     Napi::Value NativeEngine::CreateIndexBuffer(const Napi::CallbackInfo& info)
     {
         const Napi::ArrayBuffer bytes = info[0].As<Napi::ArrayBuffer>();
+        
         const uint32_t byteOffset = info[1].As<Napi::Number>().Uint32Value();
         const uint32_t byteLength = info[2].As<Napi::Number>().Uint32Value();
         const bool is32Bits = info[3].As<Napi::Boolean>().Value();
         const bool dynamic = info[4].As<Napi::Boolean>().Value();
 
+        auto reference = Napi::Persistent(info[0]);
+
         const uint16_t flags = (is32Bits ? BGFX_BUFFER_INDEX32 : 0);
-        IndexBuffer* indexBuffer = new IndexBuffer{gsl::make_span(static_cast<uint8_t*>(bytes.Data()) + byteOffset, byteLength), flags, dynamic};
+        IndexBuffer* indexBuffer = new IndexBuffer{gsl::make_span(static_cast<uint8_t*>(bytes.Data()) + byteOffset, byteLength), reference, flags, m_runtimeScheduler, dynamic};
```

Comment:
Doesn't this take a hard dependency on the location -- the raw pointer address -- of the underlying data remaining constant into the future? Is that part of the contract for all of the JS engines? Are any of them allowed to move their data?

Reply:
I think this will be related to @bghgary 's question on if it is ok to hold a reference to the JS buffer. That implies what happens when this data is overridden, release and if JS engine might move that buffer at some point. I will do some research to see if this information can be found anywhere.